### PR TITLE
[docs] Fix Next.js hosting on mui.com

### LINKED ISSUE #660: [docs] Redirection issue
### The problem

All requests to resources under `_next` when navigating to [https://mui.com/toolpad/](https://mui.com/toolpad/) fail, because of – I imagine – [this](https://github.com/mui/material-ui/blob/62a777e6ea4dd13c5e9f369b6c843a389b160dda/docs/public/_redirects#L401) conflicting rewrite rule:
Screenshot 2022-07-15 at 1 31 11 PM

### Ideas on solutions

1. Setting a custom build directory instead of `.next` might be a possible way to fix this => doesn't work it has no impact on the URLs.
2. Configuring the `_next` prefix => not possible https://github.com/vercel/next.js/issues/31718
3. Using `assetPrefix` => OK-ish, not awesome because it would break the standalone previews
4. Using `basePath` => doesn't work because it breaks the links. Links we can't change because not coming from this repository.

### PR DESCRIPTION
**Pull request by bharatkashyap**
- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

See #660 

https://deploy-preview-661--mui-toolpad-docs.netlify.app/toolpad/

### CODE REVIEW

File: docs/next.config.js [hunk 1]
```diff
@@ -141,6 +141,7 @@ module.exports = withTM({
     return map;
   },
   reactStrictMode,
+  distDir: 'dist',
```
Comment (oliviertassinari) [inline docs/next.config.js:144]: Based on the nature of the problem that we are trying to solve: scoping. I think that we should bring more context. Maybe something like this: 
```suggestion
  distDir: '_next/toolpad',
```

**Review by oliviertassinari — COMMENTED** [review]
this doesn't seem to work, the assets are still hosted under the conflicting folder: https://deploy-preview-661--mui-toolpad-docs.netlify.app/_next/static/chunks/871-bbf549ab618193c5.js

**Comment by oliviertassinari** [comment]
I suspect that we need `assetPrefix: /toolpad/`.

File: docs/next.config.js (hunk 1 above)
Comment (Janpot) [reply docs/next.config.js:144]: I think the `basePath` option is meant for exactly this purpose, scoping.

Comment (oliviertassinari) [reply docs/next.config.js:144]: https://github.com/vercel/next.js/issues/31718#issuecomment-1185690997

**Comment by bharatkashyap** [comment]
@Janpot @oliviertassinari

https://github.com/mui/mui-toolpad/pull/661/commits/b46dc1ec2f49fedb8b623a65bb1c2cdbd07c536c

 Using the `NETLIFY` environment variable to differentiate adding an `assetPrefix` along with a rewrite rule in the Toolpad docs folder for Netlify builds seems to do the trick, as the [latest preview](https://deploy-preview-661--mui-toolpad-docs.netlify.app/toolpad/) shows.

**Comment by oliviertassinari** [comment]
Ah smart. 👍 from my it. We avoid the limitation of point 3 in https://github.com/mui/mui-toolpad/issues/660.

---

- To cleanup this PR, I think that we should revert the change from `docs/.next` to `docs/dist` it doesn't help.
- For mui.com, I think that we need to be aware of:

  > By default, we limit internal rewrites to one “hop” — you cannot proxy from Netlify SiteA to Netlify SiteB to Netlify SiteC in a single request. This limitation may be amended for customers on a case-by-case basis. [Contact support](https://www.netlify.com/support) for more information.

  https://docs.netlify.com/routing/redirects/rewrites-proxies/#shadowing

  To avoid 2 hops for the JavaScript requests:
  
  - starting from https://mui.com/toolpad/_next/static/chunks/framework-1bf860015beb9572.js
  - redirected to https://mui-toolpad-docs.netlify.app/toolpad/_next/static/chunks/framework-1bf860015beb9572.js
  - to finally be resolved at https://mui-toolpad-docs.netlify.app/_next/static/chunks/framework-1bf860015beb9572.js
  
  we would need to update https://github.com/mui/material-ui/blob/62a777e6ea4dd13c5e9f369b6c843a389b160dda/docs/public/_redirects#L401 to be 
  
  ```
  /toolpad/_next/* https://mui-toolpad-docs.netlify.app/_next/:splat 200
  ```

File: docs/public/_redirects [hunk 2]
```diff
@@ -1 +1,2 @@
 / /toolpad/
+/toolpad/_next/* /_next/:splat
```
Comment (oliviertassinari) [inline docs/public/_redirects:2]: A comment could be good.

```suggestion
# Avoid conflicts with the other Next.js apps hosted under https://mui.com/
/toolpad/_next/* /_next/:splat
```

File: docs/next.config.js [hunk 3]
```diff
@@ -46,10 +46,13 @@ if (reactStrictMode) {
   console.log(`Using React.StrictMode.`);
 }
 
+const isDeployment = process.env.NETLIFY === 'true';
+
 module.exports = withTM({
   eslint: {
     ignoreDuringBuilds: true,
   },
+  assetPrefix: isDeployment ? '/toolpad' : '',
```
Comment (oliviertassinari) [inline docs/next.config.js:55]: A comment could be good.

```suggestion
  // Avoid conflicts with the other Next.js apps hosted under https://mui.com/
  assetPrefix: isDeployment ? '/toolpad' : '',
```

**Review by oliviertassinari — COMMENTED** [review]
👌

**Pull request merged**